### ISSUE TITLE
Plugging in a joystick after the game starts.

### ISSUE THREAD

**Issue by Danfun64** [chocolate-doom/chocolate-doom#653]
commit 6aad2f80343bbc527b7f652fc9c1752a34442afd precludes the possibility that someone would *want* to plug in their joystick after the game starts, or was that never possible to begin with?

**Comment by fragglet** [comment]
I agree with this sentiment, and I noticed that the SDL2 API includes the ability to dynamically detect new controllers when they are plugged in.

It's a slightly complicated problem because we only want to use a controller if it has been configured in the setup tool. The new API has the concept of a "Joystick GUID"; quote:

> The term JoystickGUID is a stable 128-bit identifier for a joystick device that does not change over time, it identifies class of the device (a X360 wired controller for example). This identifier is platform dependent.

But from this description it sounds like it isn't possible to distinguish between different controllers. For example if you have two identical controllers you can't tell the two apart from the GUID. So we need to retain the index in an advisory context so that it's possible to choose a specific controller.

With that in mind, the logic should probably look like:

* The config file specifies the GUID of the controller and an index.
* On startup, if joystick input is enabled, we search for a controller with the configured GUID. If a single controller is found with that GUID, we use it.
* If multiple controllers are found with that GUID, we pick the one that has the index=joystick_index (assuming one has that index).
* If multiple controllers are found with that GUID but none with that index, we pick the first.
* If no controllers are found with that GUID, joystick input is initially disabled, but if a new controller is plugged in with the expected GUID, we use it.

**Comment by mgorny** [comment]
Not sure if I should file a separate issue since it's kinda related: if the joystick becomes temporarily unplugged (read: shoddy cable, pulling too hard), crispy-doom doesn't pick it up once it's reconnected. Means you literally have to restart the game whenever that happens.

**Comment by fabiangreffrath** [comment]
We would have to detect `SDL_EVENT_JOYSTICK_ADDED` events and handle them accordingly in the main loop.
